sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubProducer.cs

```diff
@@ -344,6 +376,26 @@ protected EventHubProducer()
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override string ToString() => base.ToString();
 
+        private DiagnosticScope CreateDiagnosticScope()
+        {
+            DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(Azure)}.{nameof(Messaging)}.{nameof(EventHubs)}.{nameof(EventHubProducer)}.Send");
+            scope.AddAttribute("kind", "producer");
+            scope.AddAttribute("component", "eventhubs");
+            scope.AddAttribute("message_bus.destination", EventHubName);
+            scope.AddAttribute("peer.address", _endpoint);
+            scope.Start();
+
+            return scope;
+        }
+
+        private static void InstrumentMessages(IEnumerable<EventData> events)
+        {
+            foreach (var eventData in events)
```

Comment:
same here - all events have the same Id - causation is lost

Reply:
Likewise, the same additional context as above.  Each event in the batch is sent as a single AMQP message to the Event Hubs service and treated as such.

Reply:
right, but transport of these events is less of the concern than tracing individual events. It's interesting when a particular event has been sent, but only along with everything that happened with this event before/after.